x-pack/plugin/sql/qa/server/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/DataLoader.java

```diff
@@ -396,15 +454,58 @@ private static void csvToLines(String name, CheckedBiConsumer<List<String>, List
             if (titlesString == null) {
                 throw new IllegalArgumentException("[" + location + "] must contain at least a title row");
             }
-            List<String> titles = Arrays.asList(titlesString.split(","));
+            List<String> titles = splitCsvLine(titlesString);
 
             String line;
             while ((line = reader.readLine()) != null) {
-                consumeLine.accept(titles, Arrays.asList(line.split(",")));
+                List<String> values = splitCsvLine(line);
+                if (values.size() != titles.size()) {
+                    throw new IllegalArgumentException(
+                        "Values count mismatch in header [" + titles.size() + "] and line " + "[" + values.size() + "] for [" + line + "]"
+                    );
+                }
+                consumeLine.accept(titles, values);
             }
         }
     }
 
+    private static List<String> splitCsvLine(String line) {
```

Comment:
Have you checked if `org.relique.jdbc.csv` (which we use already) has this functionality?

Reply:
They do have the functionality, but don't export it.